cmd/promtool/main_test.go

```diff
@@ -78,3 +79,19 @@ func mockServer(code int, body string) (*httptest.Server, func() *http.Request)
 	}
 	return server, f
 }
+
+func TestCheckExpression(t *testing.T) {
+	// Temporarily move error output from CheckExpression calls
+	// to /dev/null, only return value is relevant
+	stderr := os.Stderr
+	defer func() { os.Stderr = stderr }()
```

Comment:
Alternatively (to the comment above) I would just remove this. I think it's ok to log something on unit test.

Reply:
It makes the test output quite confusing when testing that the error case is handled properly:

```
Error while validating: parse error at char 18: unexpected end of input inside braces
PASS
ok  	github.com/prometheus/prometheus/cmd/promtool	0.020s
```

Can ofc remove the lines and allow it, but to me that looks quite dirty :smile: . Let me know if that's still OK and I'll make it log :).